hnn_core/tests/test_gui.py

```diff
@@ -563,3 +565,68 @@ def add_child_decorated(self, to_add):
     # Check if the widgets are relinked, the selected index should be synced
     gui.tab_group_1.selected_index = 0
     assert gui.tab_group_2.selected_index == 0
+
+
+def test_gui_download_simulation():
+    """Test the GUI download simulation pipeline."""
+    gui = HNNGUI()
+    _ = gui.compose()
+    gui.params['N_pyr_x'] = 3
```

Comment:
Is there any way to set this with the `Network(..., mesh_param=(3, 3)` API? We've worked really hard to remove code that depends on modifying the `.params` object so that we can remove this attribute entirely.

Reply:
The GUI currently saves the params as an attribute and initializes a new Network object when it needs one to run a function (such as running simulations).  This is so that any connectivity or drive updates though the widgets is used in a new Network object. There isn't a Network object saved to an attribute which that can be accessed with the API call. Does that make sense? 

We could potentially change this with the new json implementation though.

Reply:
Would this be easy to tack on to that PR? I suppose it might be a whole separate PR to disentangle the params from the GUI code.